standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -1822,8 +1907,13 @@ public boolean dropPartition(String dbName, String tableName, String partName, b
   @Override
   public boolean dropPartition(String catName, String db_name, String tbl_name, String name,
                                boolean deleteData) throws TException {
-    return client.drop_partition_by_name_with_environment_context(prependCatalogToDbName(
-        catName, db_name, conf), tbl_name, name, deleteData, null);
+    DropPartitionRequest dropPartitionReq = new DropPartitionRequest();
+    dropPartitionReq.setDbName(db_name);
+    dropPartitionReq.setTblName(tbl_name);
+    dropPartitionReq.setCatName(catName);
+    dropPartitionReq.setPartName(name);
+    dropPartitionReq.setDeleteData(deleteData);
+    return client.drop_partition_by_name_req(dropPartitionReq);
```

Comment:
for the `drop_partition_by_name_req` and `drop_partition_req`, I would prefer to add only `drop_partition_req` for both purposes, Let HMSHandler check if the client sets the partition name or partition vals, the same to `append_partition_by_name_req` and `append_partition_req`

Reply:
Ack